### ISSUE TITLE
Add options to remember code snippets in conversations

### ISSUE THREAD

**Issue by mysticaltech** [danny-avila/chatgpt-clone#9]
Hello Danny, great initiative, it would be super amazing to have the ability to remember code snippets within the conversations, both those that come as input and those that are generated, as it helps chatgpt not lose the context when doing technical work. 

Currently, for instance, I have to copy and paste a thousand times, otherwise, it loses context too quickly, so I have to refresh it every time I ask a new question. Just being able to click to insert previous snippets would be awesome!

**Comment by danny-avila** [comment]
@mysticaltech I appreciate your feedback! Perhaps a right-click context menu somewhere that includes previous snippets by number? If you have any ideas where it could fit in the current UI, as I'm trying to adhere to original official style, let me know

**Comment by mysticaltech** [comment]
Yep, right-clicking with a flat (like the first 100 chars without line breaks) preview of the previous snippets and an indication if it was generated or pasted would do it for me!
